doc/auto_storage.md

```diff
@@ -564,29 +565,134 @@ or re-formatted.
 The storage proposal must make possible to define what to do with existing partitions and logical
 volumes. Even with existing MD RAIDs or LVM volume groups.
 
-In order to provide the same capabilities than the Guided proposal (see below) it must be possible
-to specify that a given partition (or LVM logical volume) must be:
+A `search` section allows to match the definition of a partition or an LVM logical volume with one
+(or several) devices existing in the system. In order to provide the same capabilities than the
+Guided proposal (see below) it must be possible to specify that a given partition or volume must be:
 
   - Deleted if needed to make space for the newly defined devices
   - Deleted in all cases
   - Shrunk to the necessary size to make space for new devices
-  - Shrunk to a given size, maybe a range, in all cases (not really possible in the current Guided
+  - Shrunk or extended to a given size, maybe a range (not really possible in the current Guided
     Proposal)
 
-It would be desirable to have the possibility of expressing some combinations of the above, like
-"try to shrink it to make space but proceed to delete it if shrinking it is not enough".
+It is even possible to express some combinations of the above, like "try to shrink it to make space
+but proceed to delete it if shrinking it is not enough".
+
+Deletion can be achieved with the corresponding `delete` flag or the alternative `deleteIfNeeded`.
+If any of those flags are active for a partition, it makes no sense to specify any other usage
+(like declaring a file system on it).
+
+The following example deletes the partition with the label "root" in all cases and, if needed, keeps
+deleting other partitions as needed to make space for the new partition of 30 GiB.
+
+```json
+"storage": {
+    "drives": [
+        {
+            "partitions": [
+                {
+                    "search": {
+                        "condition": { "property": "fsLabel", "value": "root" }
+                    },
+                    "delete": true
+                },
+                { "search": {}, "deleteIfNeeded": true },
+                { "size": "30 GiB" }
```

Comment:
NP: I would add an example for resize or delete if needed:

~~~
{
  "search": { ... }
  "deleteIfNeeded": true,
  "size": { "min": 0 }
}
~~~

Reply:
It's there, at the end of the section where it explains how to combine resizing and deleting.